lib/modules/us_core_guidance/clinicalnotes_sequence.rb

```diff
@@ -56,61 +46,121 @@ def test_clinical_notes_document_reference(category_code)
 
         resource_class = 'DocumentReference'
         patient_ids = @instance.patient_ids.split(',')
+        self.document_attachments = {} if document_attachments.nil?
+        attachments = {}
 
-        resources_found = patient_ids.any? do |patient_id|
+        patient_ids.each do |patient_id|
           search_params = { 'patient': patient_id, 'type': category_code }
 
           reply = get_resource_by_params(versioned_resource_class(resource_class), search_params)
-          assert_response_ok(reply)
-          assert_bundle_response(reply)
 
-          next unless reply&.resource&.entry&.any? { |entry| entry&.resource&.resourceType == resource_class }
-
-          self.document_attachments = ClinicalNoteAttachment.new(resource_class) if document_attachments.nil?
+          # If server require status, server shall return 400
+          # https://www.hl7.org/fhir/us/core/general-guidance.html#search-for-servers-requiring-status
+          if reply.code == 400
+            begin
```

Comment:
This code should be moved into its own method so that it isn't duplicated between in these two tests. It looks like there are already copy/paste errors.

Reply:
fixed